docs/integrations/web.md

```diff
@@ -44,60 +41,228 @@ allows you to smoothly build in the logic you need and try out various component
-* Image
+* Title
+* Description
 * Input: component that roughly corresponds to the [HTML input element](https://www.w3schools.com/html/html_form_input_types.asp), with the input data type specified by setting `type`
-* SubmitButton: basic button component, triggers a `Submit` event
 * TextArea: component for longer text input
-* Title
+* Select: choose one option from a list
+* CloudSelect: a cool multi-select component
+* Quantity select: component for specifying quantities
+* Toggle: component that imitates a switch
+* Nested Toggle: you guessed it, it's a toggle inside a toggle
+* Image
+* SubmitButton: basic button component, triggers a `Submit` event
+
+Components that are used to collect some sort of input or activity from the user (text areas, buttons, select components etc.) are referenced through the `inputId`.
+You can even make all of these components obligatory by setting its default value of the `required` field to `true`.
+If a component is set to required, users can only progress within the flow if they interact with it accordingly.  
 
-Components that are used to collect some sort of input or activity from the user (text areas, buttons etc.) are referenced through the `inputId`. For example, here is a rule that registers the user entered their name and submitted it by pressing a button:
+For example, here is a rule that registers the user entered their name and submitted it by pressing a button:
 ```java
 rule "Save user name"
     when
-        signal: Submit(buttonId == "submitName", name: getParamAsString("name")) from entry-point "signals"
+        signal: Submit(buttonId == "submit-name", firstName: getParamAsString("first-name")) from entry-point "signals"
         agent: Customer()
     then
-        modify(agent){setName(name)};
+        modify(agent){setFirstName(firstName)};
         delete (signal);
 end
 ```
 
+On each press of a `SubmitButton`, a `Submit` event is triggered. You can catch a specific `Submit` event inside a rule using the defined `buttonId`, as seen in the example above.
+This type of event also contains values of all the components that have been previously filled out. You can access them by using the `getParamAs____()` functions, which all take the `inputId` of the component as a parameter.
+
+:::info Type of the `getParamAs____()` functions
+
+- `public String getParamAsString(String param)`
+- `public Boolean getParamAsBool(String param)`
+- `public Integer getParamAsInteger(String param)`
+- `public Double getParamAsDouble(String param)`
+- `public <T> List<T> getParamAsList(String param, Class<T> cls)`
+- `public <T> Map<String, T> getParamAsMap(String param, Class<T> valueCls)`
+- `public <T, U> Map<T, U> getParamAsMap(String param, Class<T> keyCls, Class<U> valueCls)`
+- `public <T> T getParamAs(String param, Class<T> cls)`
+- `public JsonNode getParam(String param)`
+
+:::
+
+TODO - is this a buttonId or inputId?
```

Comment:
Button clicks are buttonId, and it's referenced as `buttonId == "something"`. All inputs go through the `inputId`